core/schemas/observable.py

```diff
@@ -128,8 +102,179 @@ def delete_context(
         return self.save()
 
 
-def find_type(value: str) -> ObservableType | None:
+def guess_type(value: str) -> str | None:
+    """
+    Guess the type of an observable based on its value.
+
+    Returns the type if it can be guessed, otherwise None.
+    """
     for obs_type, obj in TYPE_MAPPING.items():
-        if obj.is_valid(value):
-            return obs_type
+        if not hasattr(obj, "validate_value"):
+            continue
+        try:
+            if obj.validate_value(value):
+                return obs_type
+        except ValueError:
+            continue
     return None
+
+
+def create(*, value: str, type: str | None = None, **kwargs) -> ObservableTypes:
+    """
+    Create an observable object without saving it to the database.
+
+    value argument representing the value of the observable.
+
+    if kwargs does not contain a "type" field, type will be automatically
+    determined based on the value. If the type is not recognized, an observable
+    of type generic will be created.
```

Comment:
Please adjust docstring here

Reply:
fixed in 900075a392262e00e39fc80decbae7eda306ac94